core/src/container/config.rs

```diff
@@ -0,0 +1,97 @@
+extern crate serde_derive;
+extern crate toml;
+use holochain_agent::{Agent, Identity};
+use holochain_core_types::error::{HcResult, HolochainError};
+use holochain_dna::Dna;
+use serde::Deserialize;
+use std::{fs::File, io::prelude::*};
+
+pub struct StorageConfiguration {}
+
+#[derive(Deserialize, Clone)]
+pub struct AgentConfiguration {
+    id: String,
+    key_file: Option<String>,
+}
+
+impl Into<Agent> for AgentConfiguration {
+    fn into(self) -> Agent {
+        Agent::new(Identity::new(self.id))
+    }
+}
+
+#[derive(Deserialize)]
+pub struct DNAConfiguration {
+    id: String,
+    file: String,
+    hash: String,
+}
+
+impl Into<HcResult<Dna>> for DNAConfiguration {
+    fn into(self) -> HcResult<Dna> {
+        let mut f = File::open(self.file)
+            .map_err(|_| HolochainError::IoError(String::from("Could read from file")))?;
+        let mut contents = String::new();
+        f.read_to_string(&mut contents)
+            .map_err(|_| HolochainError::IoError(String::from("Could read from file")))?;
+        Dna::from_json_str(&contents)
+            .map_err(|_| HolochainError::IoError(String::from("Could not create dna form json")))
+    }
+}
+
+#[derive(Deserialize)]
+pub struct InstanceConfiguration {
+    id: String,
+    #[serde(rename = "DNA")]
+    dna: String,
+}
+
+#[derive(Deserialize)]
+pub struct Configuration {
+    #[serde(rename = "Agents")]
```

Comment:
I would actually call that one `agent`. Because in toml, you'll be able declare an agent configuration by saying:

```toml
[[agent]]
property = "value"
...

[[agent]]
property = "value"
...

[[agent]]
property = "value"
...


```

It's the same with the `Cargo.toml`. They use singular for lists to make items more intuitive as in: "I'm describing an agent.... now another agent"

Reply:
Also if this is optional anyway, you might wanna add `#[serde(default)]` to the field. This way you can drop the `Option<T>`